### ISSUE TITLE
DELETE FROM in iOS incorrectly UPDATES wrong column, resulting in "no such column" exception

### ISSUE THREAD

**Issue by MarcoG3** [capacitor-community/sqlite#445]
**Describe the bug**
When deleting an entry from `Table_1` which has a `FOREIGN KEY` to `Table_2`, on iOS an exception occurs.

**To Reproduce**

```
CREATE TABLE IF NOT EXISTS Table_1 (
      id INTEGER PRIMARY KEY NOT NULL,
      result_id TEXT,
      result_slug TEXT,

      sql_deleted BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1)),
      last_modified INTEGER DEFAULT (strftime('%s', 'now')),

      FOREIGN KEY (result_id, result_slug) REFERENCES Table_2(id, slug) ON DELETE SET DEFAULT
    );
```

```
CREATE TABLE IF NOT EXISTS Table_2 (
      id TEXT NOT NULL,
      slug TEXT NOT NULL,

      sql_deleted BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1)),
      last_modified INTEGER DEFAULT (strftime('%s', 'now')),

      PRIMARY KEY (id, slug)
    );
```

I've debugged through XCode what's happening and
- On executing `DELETE FROM Table_1 WHERE id = ?`
- `findReferencesAndUpdate` gets called, and the reference columns are incorrectly calculated
- (line 576, UtilsSQLCipher.swift) - `stmt` gets calculated incorrectly as `UPDATE Table_2 SET sql_deleted = 1 WHERE result_id =  ?` instead of `UPDATE Table_2 SET sql_deleted = 1 WHERE id =  ?`.
- Basically, it uses the column of `Table_1` to update reference of `Table_2`.

**Screenshots**
<img width="830" alt="image" src="https://github.com/capacitor-community/sqlite/assets/4958742/a4c37e6b-30bb-49a9-8a7e-6b9ab0d512d2">

**Smartphone (please complete the following information):**

- Device: iPhone 13
- OS: iOS 16

**Comment by MarcoG3** [comment]
Yes. I am trying to delete from `Table_1`, and the current logic is trying to delete (using the wrong column) an entry which should not be even deleted in the first place from `Table_2`.

The opposite should be true; since I have a foreign key in `Table_1` referencing some columns of `Table_2`, only when I delete an entry from `Table_2` it should proceed, eventually, with deleting row(s) from `Table_1`.

Thank you for looking into this!

**Comment by jepiqueau** [comment]
@MarcoG3 as you have a combined primary key on table if i delete using the primary key (id,slug) it is also failing so i have to review all this again it will also failed on the android, electron, web plugin so that will take some time before i solved this. Sorry for this.

**Comment by jepiqueau** [comment]
@MarcoG3 i wonder if the foreign key should not be defined on table_2 and not table_1

**Comment by MarcoG3** [comment]
Yeah, I just checked on web and it is failing there as well. In my case it would not make any sense to define the foreign key into `Table_2` - the reason why I added the foreign key in the first place (and set `ON DELETE SET DEFAULT`) is because I need the consistency between the two. For example, if we `DELETE FROM Table_2 WHERE id=33 AND slug='blabla'` and we have a row in `Table_1` that has `result_id=33` and `slug='blabla'` then it would make sense for the `ON DELETE SET DEFAULT` to fire and set those as `NULL`.

**Comment by MarcoG3** [comment]
package.json

```
"@capacitor-community/sqlite": "^5.0.5-2",
"jeep-sqlite": "^2.3.3",
"react-sqlite-hook": "^4.0.0",
```


```
  CREATE TABLE IF NOT EXISTS Table_1 (
      id INTEGER PRIMARY KEY NOT NULL,
      product_id TEXT NOT NULL,
      result_id TEXT,
      result_slug TEXT,

      sql_deleted BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1)),
      last_modified INTEGER DEFAULT (strftime('%s', 'now')),

      FOREIGN KEY (result_id, result_slug) REFERENCES Quiz_Results(id, slug) ON DELETE SET DEFAULT,
      FOREIGN KEY (person_id) REFERENCES People(id) ON DELETE SET DEFAULT
    );


CREATE TABLE IF NOT EXISTS Table_2 (
      id TEXT NOT NULL,
      slug TEXT NOT NULL,

      sql_deleted BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1)),
      created_at INTEGER NOT NULL,
      last_modified INTEGER DEFAULT (strftime('%s', 'now')),

      PRIMARY KEY (id, slug),
      FOREIGN KEY (person_id) REFERENCES People(id)
);
```

What I am doing is actually:

`DELETE FROM Table_1 WHERE product_id = 1234`

What happens is that a query like

`UPDATE Table_2 SET result_id = ?` is being run from `findReferencesAndUpdate` (in iOS)

Let me know if I can expand further or if you are still unable to replicate the issue.

**Comment by jepiqueau** [comment]
@MarcoG3 if i look at findReferencesAndUpdate i got this 
```
Printing description of refTable:
"Table_2"
Printing description of withRefsNames:
▿ 2 elements
  - 0 : "result_id"
  - 1 : " result_slug"
Printing description of colNames:
▿ 2 elements
  - 0 : "id"
  - 1 : " slug"
Printing description of colNames:
▿ 2 elements
  - 0 : "id"
  - 1 : " slug"
Printing description of uWhereStmt:
"WHERE result_id =  1"
Printing description of updTableName:
"Table_1"
Printing description of stmt:
"UPDATE Table_1 SET sql_deleted = 1 WHERE result_id =  1"
Printing description of stmt:
"UPDATE Table_1 SET sql_deleted = 1 WHERE result_id =  1"

```

**Comment by jepiqueau** [comment]
@MarcoG3 Can you debug and give me the following when you enter in findReferencesAndUpdate
- tableName
- whereStmt
- values
and going step by step
- references
- tableNameWithRefs 
- refTable
- withRefsNames
- colNames
- uWhereStmt
- updTableName
- updColNames
- stmt

i assume it is returning an error from prepareSQL

**Comment by MarcoG3** [comment]
* tableName: `Table_1`
* whereStmt: `WHERE product_id = ? AND product_type = ?` (I am not looking up on `PRIMARY` key here, which is what it seems you are doing in your tests, not sure if that matters)
* values: `[123123, "book"]`
* references: 
```
[
   "(result_id, result_slug) REFERENCES Table_2(id, slug)", 
   "(person_id) REFERENCES People(id)"
]
```

Forgot to mention that I have two FOREIGN KEYS, also not sure if that matters or not

* tableNameWithRefs: `Table_1`
* refTable: `Table_2`
* withRefsNames: `["result_id", "result_slug"]`
* colNames: `["id", "slug"]`
* uWhereStmt: `WHERE result_id = ?` (here I think is where it messes it up, because `Table_2` has no `result_id`)
* updTableName: `Table_1`
* updColNames: `["id", "slug"]`
* stmt: `UPDATE Table_2 SET sql_deleted = 1 WHERE result_id =  ?`

I have updated to the latest version of the plugin.

Thanks!

**Comment by jepiqueau** [comment]
@MarcoG3 i cannot work like this i am spending quite a time for your issue and i still do not have the complete picture of what you are doing so please give me that picture with a complete code that i am able to reproduce it. Share your code or a simple demo reproducing the issue otherwise i will move on something else

**Comment by MarcoG3** [comment]
@jepiqueau kindly take a look at https://github.com/MarcoG3/capacitor-sqlite-issue445-poc 

let me know if I can help in other ways.

**Comment by MarcoG3** [comment]
https://github.com/MarcoG3/capacitor-sqlite-issue445-poc/blob/main/src/components/TestIssue445.tsx

**Comment by jepiqueau** [comment]
@MarcoG3 You are running a really old version of @capacitor/core & @capacitor-community/sqlite As i already suggested update to the latest release 5.0.6 so at least we are on the same path

**Comment by MarcoG3** [comment]
I have upgraded @capacitor/core to v5 and @capacitor-community/sqlite to v5.0.6 and still the same error is shown

**Comment by jepiqueau** [comment]
@MarcoG3 I am digging into it
The problem is that values are not correct

Printing description of stmt:
"UPDATE Table_2 SET sql_deleted = 1 WHERE result_id =  ?"
Printing description of selValues:
0 elements

**Comment by jepiqueau** [comment]
@MarcoG3 thank for providing the code it helps to identify the issue. I have now to solve it which will take me some times .

**Comment by MarcoG3** [comment]
Hi @jepiqueau, thanks for taking a look into it. Any update on the issue?

**Comment by jepiqueau** [comment]
@MarcoG3 i am working on it is taking as i saw that the code was not correct and did not take into account the ON DELETE CASCADE,  RESTRICT, SET DEFAULT mode properly. So i have to find the way to do it. I am testing it on the jeep-sqlite first and will move on iOS and after Electron and Android. I will publish first a jeep-sqlite new version so you can test it and tell me if it works for your case and later an intermediate version of the @capacitor-community/sqlite with the iOS part only. Does that fit with you. Mid next week i hope sorry for the inconvenience.i also take the opportunity to refresh the code.

**Comment by MarcoG3** [comment]
That sounds perfect to me - I'll gladly test the new version when you release it, thanks!

**Comment by jepiqueau** [comment]
@MarcoG3 this is now fixed in `jeep-sqlite@2.4.0` if you want to test it, please report any issues. I will now concentrate on the iOS part of the `@capacitor-community/sqlite` issue this preliminary release and then go for Android and Electron

**Comment by MarcoG3** [comment]
I just tested on latest `jeep-sqlite@2.4.0` and I confirm you that it works perfectly now. Looking forward the updates on both iOS and Android, thank you! I appreciate your work.

**Comment by jepiqueau** [comment]
@MarcoG3 iOS is now available in release 5.0.7-1

**Comment by MarcoG3** [comment]
I will test in the following days and let you know!

**Comment by MarcoG3** [comment]
@jepiqueau I confirm that the issue has been resolved on iOS by upgrading to `@capacitor-community/sqlite@5.0.7-1`.

In the next few days I'll also be able to test the Android bug fix release whenever it is ready.

**Comment by jepiqueau** [comment]
@MarcoG3 thanks for the feedback i did the Electron one and have not start yet the Android one. I have seen already some improvments to do allowing for <>,<=, >,>= operators and also manage not only the AND but also the OR and NOT so still some work as you see

**Comment by jepiqueau** [comment]
@MarcoG3 it is also fix for Android in release 5.0.7 so i close the issue
